src/crimson/os/seastore/onode_manager/staged-fltree/stages/stage_types.h

```diff
@@ -433,10 +433,30 @@ struct node_stats_t {
   unsigned num_kvs = 0;
 };
 
+} // namespace crimson::os::seastore::onode
+
+namespace std {
+template <crimson::os::seastore::onode::match_stage_t STAGE>
+std::ostream& operator<<(
+    std::ostream& os,
+    const crimson::os::seastore::onode::staged_position_t<STAGE>& pos) {
+  return os << pos.fmt_print();
 }
+} // namespace std
 
-#if FMT_VERSION >= 90000
+
+namespace fmt {
 template <crimson::os::seastore::onode::match_stage_t S>
-struct fmt::formatter<crimson::os::seastore::onode::staged_position_t<S>> : fmt::ostream_formatter {};
-template <> struct fmt::formatter<crimson::os::seastore::onode::MatchHistory> : fmt::ostream_formatter {};
+struct formatter<crimson::os::seastore::onode::staged_position_t<S>> {
+  constexpr auto parse(fmt::format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const crimson::os::seastore::onode::staged_position_t<S>& k,
+              FormatContext& ctx) const {
+    return fmt::format_to(ctx.out(), "{}", k.fmt_print());
+  }
+};
```

Comment:
Looks `staged_position_t` cannot be automatically tagged by common/fmt_common.h

Reply:
There seem to be a problem with partial specialization.
I'll try to give it another look.